Mathlib/RingTheory/DedekindDomain/FiniteAdeleRing.lean

```diff
@@ -336,9 +336,31 @@ instance : Algebra R (FiniteAdeleRing R K) :=
 instance : IsScalarTower R K (FiniteAdeleRing R K) :=
   IsScalarTower.of_algebraMap_eq' rfl
 
+/-- Coercion `FiniteAdeleRing R K → K_hat R K`. -/
+@[coe]
+def coe (x : FiniteAdeleRing R K) : K_hat R K := x.1
```

Comment:
No worries for the question. I am far from an expert on coercions myself so I began by replicating other coercion examples I have seen and used (specifically, line 128 in Mathlib/Algebra/Quaternions) - here they had both def and instance, so this is why I added the definition. But upon more looking I think you are right that the definition is unnecessary (that is it adds nothing new) in this case as the instance introduces it as a function (unlike in the quaternion file). @[coe] was added as I presumed that is what allows the delaborator to use the `\u` notation - the tooltip for it also says it should be included in instances like we have, so maybe someone with more knowledge could explain how it already works without the tag?

Reply:
I would be happy to remove this definition in a new commit, unless you think it is idiomatic to keep it and change the instance so that it matches the style used in the quaternion file. I am unsure what the standard preferred convention is?